main.js

```diff
@@ -1,82 +1,158 @@
-// use the variables below to run Math function
-let val1;
-let op;
+let numInputs = [];
+let opInputs = [];
 
+//Define numbers and operators for calculation
 function inputNum(number) {
-    // if value is 0 then it is replaced by selected value. Otherwise adds to display value
-    if (document.getElementById('calc').value == 0 || isNaN(document.getElementById('calc').value)) {
-        if (document.getElementById('calc').value == 0 + '.') {
-            // nested if to account for 0. being treated as NaN
-            document.getElementById('calc').value = document.getElementById('calc').value + number;
-        } else {
-            document.getElementById('calc').value = number;
-        }
+    let currentNumber = document.getElementById('calc').value;
+    //debugger;
+    if ( currentNumber.indexOf('.') !== -1 ) {
+        currentNumber = currentNumber + number;
+        numInputs[numInputs.length - 1] = currentNumber;
+    } else if ( currentNumber === '0' ) {
+        currentNumber = number;
+        numInputs.push(currentNumber);
     } else {
-        document.getElementById('calc').value = document.getElementById('calc').value + number;
+        currentNumber = currentNumber + number;
+        numInputs[numInputs.length - 1] = currentNumber;
     }
+    document.getElementById('calc').value = currentNumber;
+    createHist();
+    clearOp();
 }
 
-function allClearNum() {
-    // clears input to 0 and clears all variables
-    document.getElementById('calc').value = 0;
-    val1 = undefined;
-    val2 = undefined;
-    op = undefined;
+function inputOp(op) {
+    let currentOp = document.getElementById('op').value;
+
+    if ( currentOp === '' ) {
+        opInputs.push(op);
+    } else {
+        opInputs[opInputs.length - 1] = op;
+    } 
+    document.getElementById('op').value = op;
```

Comment:
If you want here you could check if `op === '/'` and then set the element value to `&#247;` so that the UI shows the division sign consistently.

Reply:
This would only allow the user to see the &#247; when they click the button, once it is added to the history it will be replaced with /. If I change it everywhere then it will force a more complicated calculation function. I am fine with /.

Reply:
Perfect! Sounds good to me